engine/src/main/java/org/pentaho/di/trans/DefaultTransManager.java

```diff
@@ -0,0 +1,43 @@
+package org.pentaho.di.trans;
+
+import org.pentaho.di.core.exception.KettleException;
+import org.pentaho.di.core.logging.LogChannelInterface;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class DefaultTransManager implements TransMananger {
+
+  private static DefaultTransManager instance;
+
+  private Map<String, TransFactory> mapConfigurationTypeTransFactory;
+
+
+  private DefaultTransManager(){
+    mapConfigurationTypeTransFactory = new HashMap<>();
+  }
+
+  public static synchronized DefaultTransManager getInstance() {
+    if ( null == instance ) {
+      instance = new DefaultTransManager();
+    }
+    return instance;
+  }
+
+  @Override
```

Comment:
javadoc comments for DefaultTransManager#createTrans and other functions

Reply:
addressed